File: java/langchain4j/src/main/java/com/llmagents/OllamaConfig.java
package com.llmagents;

import com.google.gson.Gson;
import com.google.gson.JsonObject;

import java.nio.file.Files;
import java.nio.file.Path;

public record OllamaConfig(String host, String baseUrl, String apiKey, String defaultModel) {
    private static final Gson GSON = new Gson();

    public static OllamaConfig load(Path projectRoot) throws Exception {
        String json = Files.readString(projectRoot.resolve("config").resolve("ollama.json"));
        JsonObject o = GSON.fromJson(json, JsonObject.class);
        return new OllamaConfig(
                o.get("host").getAsString(),
                o.get("base_url").getAsString(),
                o.get("api_key").getAsString(),
                o.get("default_model").getAsString()
        );
    }
}

File: dotnet/src/TopicResearchAgent/OllamaConfig.cs
using System.Text.Json;

namespace TopicResearchAgent;

public sealed class OllamaConfig
{
    public string Host { get; init; } = "http://127.0.0.1:11434";
    public string BaseUrl { get; init; } = "http://127.0.0.1:11434/v1";
    public string ApiKey { get; init; } = "ollama";
    public string DefaultModel { get; init; } = "nemotron-3-super:cloud";

    public static OllamaConfig Load(string projectRoot)
    {
        var path = Path.Combine(projectRoot, "config", "ollama.json");
        var json = File.ReadAllText(path);
        using var doc = JsonDocument.Parse(json);
        var root = doc.RootElement;
        return new OllamaConfig
        {
            Host = root.GetProperty("host").GetString() ?? "http://127.0.0.1:11434",
            BaseUrl = root.GetProperty("base_url").GetString() ?? "http://127.0.0.1:11434/v1",
            ApiKey = root.GetProperty("api_key").GetString() ?? "ollama",
            DefaultModel = root.GetProperty("default_model").GetString() ?? "qwen2.5:7b",
        };
    }
}
